### ISSUE TITLE
Add filter to change the amount of reserved stock minutes.

### ISSUE THREAD

**Issue by morganhvidt** [woocommerce/woocommerce#27034]
https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/a404e5b24814240f15e50aa4f983b787bb9b36f1/src/StoreApi/Routes/Checkout.php#L307

10 minutes is hardcoded when reserving stock for orders. It would be great to have have a filter to change the amount of minutes.

Thanks!

**Comment by nerrad** [comment]
Note, as an update, the functionality being mentioned in this issue is now in WooCommerce Core. Any filter, if added would likely be added around here: https://github.com/woocommerce/woocommerce/blob/0b3900b84a16a819d2e5e25eab73fb81df0c8547/src/Checkout/Helpers/ReserveStock.php#L69

**Comment by ObliviousHarmony** [comment]
While WooCommerce Core's usage of the `ReserveStock` class does indeed use the `woocommerce_hold_stock_minutes` option the Checkout block uses a default of 10 minutes. There's an [issue open in that repository](https://github.com/woocommerce/woocommerce-gutenberg-products-block/issues/2882) now if you'd like to leave some comments on your take for a solution.

I've been thinking about the Core side of this and I'd like to know what kinds of use cases you had in mind for being able to customize this @morganhvidt. The actual usage of this stock reservation system is very narrow in that you need to have an order to interact with it. We're using the `woocommerce_hold_stock_minutes` option because it was originally only a system for handling stock race conditions. While I can see some ways that temporary stock reservations could be useful, the need for an order object is limiting in most of those cases.

**Comment by morganhvidt** [comment]
Thank you @nerrad for opening it back up here 😊

@ObliviousHarmony [I've added my comment to the other issue.](https://github.com/woocommerce/woocommerce-gutenberg-products-block/issues/2882#issuecomment-659814643) Please read that. I'm sorry if I wrote it in the wrong place I got a little lost with these issues.

In response to  `woocommerce_hold_stock_minutes` in core. I do believe this needs to be strictly for canceling unpaid orders. Otherwise I predict people will be limited and confused by the option. It's esstentially 1 option to serve 2 different uses.

I've put together a rough screenshot of what I'd suggest for the future. (I'm assuming that the Checkout Block will be added to care as a preferred checkout)
![proposed-stock-changes](https://user-images.githubusercontent.com/10958823/87746350-ed115200-c822-11ea-9e7e-f052e6ca97ec.png)

**Comment by nerrad** [comment]
Thanks Morgan, 

It is becoming apparent that there are different use-cases for Reserve Stock as you've noted - this is definitely something we'll have to work out as a team and something on our radar to deliberate. 

Your suggestion regarding options is a good one too, I'm not sure we'll tie order status to it but there definitely are different contexts that could be configurable (as noted between reserved stock when in the checkout block flow, vs stock reserved until paid).

It's one of the reasons why we're hesitant to just add filters at the moment because this new logic is still fresh and there may be some more adjustments we need to make (especially in light of [your comment](https://github.com/woocommerce/woocommerce-gutenberg-products-block/issues/2882#issuecomment-659814643) in the other issue which introduces some additional things to consider). So committing to a public API prematurely could lock us into a bad approach.

**Comment by morganhvidt** [comment]
I fully agree with what you've said. It's amazing how my different businesses are running WooCommerce in different ways. The flexibility to bend WooCommerce to fit any business model is a huge selling point.

We might need to step back and analyze the whole picture, set a direction of these features, and how reserved stock/blocks will be integrated into core. Draft orders & stock reservation can be a powerful feature for the platform.

I'd love to peak in and provide feedback on the wider scope. Let me know how I can join If these conversations are taking place elsewhere. 😊

**Comment by ObliviousHarmony** [comment]
Hello @morganhvidt,

I do think the fact that this class is being used immediately after release demonstrates a need for a more generic means of temporarily withholding stock without an order. I'm thinking we should make this table polymorphic and adjust the class so it can support holding stock for other contexts.

**Comment by ObliviousHarmony** [comment]
Hello @morganhvidt,

It's on my list to refactor this class & table to support operating without the order dependency. I'm going to mark this as an enhancement request and put it on my list. Feel free to submit a PR if you'd like though!

**Comment by morganhvidt** [comment]
Hi @ObliviousHarmony 

Awesome to hear that it's on the list! I've created a draft PR to get us started. 
Let me know your thoughts on the direction. 
PR: https://github.com/woocommerce/woocommerce/pull/27404

**Comment by Sn00z389** [comment]
This is a great feature that needs to be implemented in Woocommerce, since it is done for OpenCart and Magento. We even have a client who wants to use Woo but cannot since the cart reservation is not possible there right now.

**Comment by morganhvidt** [comment]
Hi @ObliviousHarmony Just wanted to ask if you've had time to check out my PR: #27404 
Let me know your feedback, I'm happy to help and get the ball rolling on these stock improvements.

**Comment by jamesckemp** [comment]
This issue has been automatically closed as part of the backlog cleanup announced in [Refining how we manage the WooCommerce backlog](https://developer.woocommerce.com/2026/05/29/backlog-cleanup/). It met the criteria outlined there. If this still affects a current version of WooCommerce, please follow the steps in that post. A fresh report with current reproduction steps is the fastest path to triage. If this is a feature request, please add it to the [WooCommerce feature request board](https://woocommerce.com/feature-requests/woocommerce/) so it can be considered alongside other requests.
